backend/app/services/google_scraper.rb

```diff
@@ -1,36 +1,38 @@
 # frozen_string_literal: true
 
 class GoogleScraper < ApplicationService
-  attr_reader :keywords
-  attr_reader :user_id
+  attr_reader :keyword
 
   WEB_BASE_URL = ENV['WEB_BASE_URL']
   HEADERS      = {
     "Accept-Language" => "en-US",
     "User-Agent"      => "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/112.0.0.0 Safari/537.36"
   }
 
-  def initialize(keywords, user_id)
-    @user_id = user_id
-    @keywords = keywords
+  def initialize(keyword)
+    @keyword = keyword
   end
 
   def call
-    @keywords.map do |keyword|
-      report = initial_value()
-      query = {
-        q: keyword,
+    report = initial_value()
+    query = {
+        q: @keyword,
         hl: 'en'
-      }
+    }
 
+    begin
       conn = Faraday.new(url: WEB_BASE_URL, headers: HEADERS)
       response = conn.get('search?', query)
       html = response.body
-      sleep 1
-      doc = Nokogiri::HTML(html)
 
-      # keyword
-      report[:word] = keyword
+      if response.status == 302
+        report[:status] = :failed
+        
+        return report
+      end
+
+      sleep 5
```

Comment:
Using sleep is code smell 👃 Google does have rate limiting in place. So part of the challenge is not to get blocked by Google. A simple appraoche is to add randomness in scheduling a scraping job.:

```
ScrapingJob.set(wait: rand(60).seconds).perform_later(self)
```

Reply:
I've been struggling from get blocked by Google and I don't know how long it take.
I have tried this approach already.
it works so well kub.